### ISSUE TITLE
The `Insert` API now only queries Accounts internally at most once regardless of the quantity of RawContacts being inserted

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#332]
## Problem

Internally, the `Insert` API queries the Accounts in the system to check if the RawContact's Account matches one of those in the system **for every RawContact**,

> ℹ️ The following is pseudocode.

```kotlin
for (rawContact in rawContacts) {
    val accountsList = getAccountsList()
    val isValidAccount = accountsList.contains(rawContact.account)
    ...
}
```

This slows down total insertion speed of multiple RawContacts, especially as the number of RawContacts being inserted increases.

## Solution

Query Accounts once per insert operation (on `commit`) instead of for every RawContact in a for-loop,

> ℹ️ The following is pseudocode.

```kotlin
val accountsList = getAccountsList()
for (rawContact in rawContacts) {
    val isValidAccount = accountsList.contains(rawContact.account)
    ...
}
```

**Comment by vestrel00** [comment]
This was closed in https://github.com/vestrel00/contacts-android/commit/67e1d0b0d98d71e2a36e46e40fb3cc82170df088#r128739780

**Comment by vestrel00** [comment]
This is included in release [0.3.2](https://github.com/vestrel00/contacts-android/releases/tag/0.3.2).
